### DISCUSSION TITLE
mysql 单表CDC 2.02 Table API连接正常 2.1.1&2.2.1 连接异常 Process finished with exit code -17

### DISCUSSION THREAD

**Discussion by HANAPolo12** [apache/flink-cdc#1209] [Q&A] [answered]
刚入行不久，烦请大神指点代码问题，感谢

mysql : MariaDB 10.4
flink : 1.13.2
scala: 2.12
在 flink-connector-mysql-cdc 2.2.1&2.1.1 出现问题，程序结束Process finished with exit code -17，没有输出，但在2.02版本正常

import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment
import org.apache.flink.table.api.bridge.scala.StreamTableEnvironment
import org.apache.flink.table.api.{EnvironmentSettings, TableEnvironment}

object Mysql_CDC {
  def main(args: Array[String]): Unit = {

val bsEnv = StreamExecutionEnvironment.getExecutionEnvironment
    bsEnv.setParallelism(1)
    val bsSettings = EnvironmentSettings.newInstance().useBlinkPlanner().inStreamingMode().build()
    val tableEnvironment = StreamTableEnvironment.create(bsEnv,bsSettings)

//     创建表，connector使用mysql-cdc
    tableEnvironment.executeSql("CREATE TABLE StationLog1_copy1 " +
      "(ID INT NOT NULL, sid STRING, call_out STRING, call_in STRING, call_type STRING, call_time BIGINT, duration INT  , PRIMARY KEY (ID) NOT ENFORCED) WITH " +
      "('connector' = 'mysql-cdc', " +
      "'hostname' = 'hadoop6', " +
      "'port' = '3306', " +
      "'username' = 'root', " +
      "'password' = '', " +
      "'database-name' = 'StationLog', " +
      "'table-name' = 'StationLog1_copy1')"
      )
//'scan.startup.mode' = 'latest-offset' initial "'scan.startup.mode' = 'initial' )"

 // 创建下游数据表，这里使用print类型的connector，将数据直接打印出来 
    tableEnvironment.executeSql("CREATE TABLE sink_table (ID INT NOT NULL, sid STRING, call_out STRING, call_in STRING, call_type STRING, call_time BIGINT, duration INT ) " +
      "WITH ('connector' = 'print')")

    // 将CDC数据源和下游数据表对接起来
    tableEnvironment.executeSql("INSERT INTO sink_table SELECT * FROM StationLog1_copy1")

  }
}

**Comment by shadon178** [comment, marked as answer]
官方好像没有说支持MariaDB 
![image](https://user-images.githubusercontent.com/26808058/170178084-a0534f2a-ecf5-4b3d-94e3-f1b45bffbc24.png)

**Reply by HANAPolo12** [reply]
@shadon178 感谢，问题已解决，最新的2.2可以支持mariaDB，造成原因是maven引入了这个包就会出现上述问题

        <dependency>
            <groupId>org.apache.kafka</groupId>
            <artifactId>kafka-clients</artifactId>
            <scope>provided</scope>
            <version>2.1.1</version>
        </dependency>
